sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyUtil.java

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp;
+
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.util.annotation.NonNull;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.util.regex.Pattern;
+
+/***
+ * Utility class to help with {@link ProxyOptions}.
+ */
+public final class ProxyUtil {
+
+    private static final Pattern HOST_PORT_PATTERN = Pattern.compile("^[^:]+:\\d+");
+    private static final ClientLogger LOGGER = new ClientLogger(ProxyUtil.class);
+
+    private ProxyUtil() {
+        // private default constructor
+    }
+
+    /***
+     * Static method to construct {@link ProxyOptions} using the values from configuration .
+     *
+     * @param configuration - The client library configuration
+     * @return the constructed {@link ProxyOptions} object
+     */
+    public static ProxyOptions getDefaultProxyConfiguration(@NonNull Configuration configuration) {
```

Comment:
We generally don't use this annotation provided by Reactor. We should instead do a null check inside the method itself.

Reply:
Sounds good, added the check